src/db.c

```diff
@@ -1529,31 +1530,41 @@ int keyIsExpired(redisDb *db, robj *key) {
  * is via lookupKey*() family of functions.
  *
  * The behavior of the function depends on the replication role of the
- * instance, because slave instances do not expire keys, they wait
- * for DELs from the master for consistency matters. However even
- * slaves will try to have a coherent return value for the function,
- * so that read commands executed in the slave side will be able to
- * behave like if the key is expired even if still present (because the
- * master has yet to propagate the DEL).
```

Comment:
i think parts of that comment you delete are still wanted (good description of the scenario of a DEL that didn't yet arrive).
maybe you can revive that whole block and add the part describing force_delete_expired just below it?

Reply:
Done.